types/store/lhs.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export type LhsViewState = {
+    isOpen: boolean;
+
+    // Static pages (e.g. Threads, Insights, etc.)
+    currentStaticPageId: string;
+}
+
+export enum LhsItemType {
+    None = 'none',
+    Page = 'page',
+    Channel = 'channel',
+}
+
+export enum LhsPage {
```

Comment:
I think in a separate ticket we could update the channel switcher (cmd+K) to use these new enums rather than their current approach of `Constants.THREADS` & `Constants.INSIGHTS`. Unify the channel switcher with this in a way so that it's easier for people making future additions to the LHS.

https://github.com/mattermost/mattermost-webapp/pull/11138/files my old channel switcher PR for insights for context